File: ch01/ex1_37_a.py
#! /usr/bin/env python3


import numpy as np
import matplotlib.pyplot as plt


# WMS Ex. 1.37 (a) - Relative frequency histogram representing
# number of foxes with certain parasites.


# Input fox-parasite datat to vector
fp = [0] * 69
fp += [1] * 17
fp += [2] * 6
fp += [3] * 3
fp += [4]
fp += [5] * 2
fp += [6]
fp += [8]


# Create the histogram.
fig, ax = plt.subplots()
ax.hist(fp, bins=range(10), weights=np.zeros_like(fp) + 1./len(fp))
ax.set_title("Number of Parasites in Foxes")
ax.set_xlabel("Number of Parasites")
ax.set_ylabel("Relative Frequency")


# Save figure and show.
plt.show()
fig.savefig("ex1_25_b.pdf", dpi=300, format='pdf')

File: ch01/ex1_37_a.r
# WMS Ex. 1.37 (a) - Relative frequency histogram representing
# number of foxes with certain parasites.

library(ggplot2)

# Input fox-parasite data to vector
fp <- c(rep(0, 69), rep(1, 17), rep(2, 6), rep(3, 3), 4, rep(5, 2), 6, 8)

# Create the histogram.
ggplot(data = data.frame(x = fp)) +
  geom_histogram(
        aes(x, y = after_stat(density), binwidth = 1, fill = "white", color = "black"
    ) +
  ggtitle("Number of Parasites in Foxes") +
  xlab("Number of Parasites") +
  ylab("Relative Frequency") +
  scale_y_continuous(labels = scales::percent) +
  theme_classic()

# Save figure and show.
ggsave("ex1_37_a.pdf", dpi = 300)
